main/lsp/LSPIndexer.cc

```diff
@@ -287,10 +287,24 @@ void LSPIndexer::transferInitializeState(InitializedTask &task) {
     task.setKeyValueStore(std::move(this->kvstore));
 }
 
-void LSPIndexer::initialize(IndexerInitializationTask &task) {
+void LSPIndexer::initialize(IndexerInitializationTask &task, std::vector<std::shared_ptr<core::File>> files) {
```

Comment:
should we take by rvalue reference?

Reply:
We could, though I tend to leave the decision about moving to call sites. In this case, given that there's exactly one call site, maybe it's the right choice to ensure that we don't accidentally copy the vector.

Reply:
That strikes me as a surprising general principle. I would expect to see every vector taken by rval ref or by reference (or by span), with every by-copy vector documented why it has to be copied.

Reply:
Just to be clear, I meant that maybe the rvalue reference is the right choice here.

In general, I like passing by value because it gives the option of copying or moving, which is often pretty situational. This is definitely a flexible principle, and for functions with a single caller I think that it's best to do whatever makes the most sense there.